server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -1591,6 +1600,24 @@ private Node stop() {
     // In this case the process will be terminated even if the first call to close() has not finished yet.
     @Override
     public synchronized void close() throws IOException {
+        CountDownLatch terminationHandlersComplete = new CountDownLatch(terminationHandlers.size());
+        terminationHandlers.forEach(handler -> {
+            // This AtomicBoolean is used to ensure one plugin calling its runnable over and over won't run down the CountDownLatch early
```

Comment:
out of curiosity, when can this happen that one plugin is running continuously (over and over) the handleTermination and it's runnable?

Reply:
I'll be removing this mechanism in favor of verifying that we have only one of these handlers anyway. This is more of a guard against anything doing it accidentally rather than something I expect to actually happen - if I were keeping this mechanism here I'd probably put an `assert` here to make it clear that multiple calls are not expected.